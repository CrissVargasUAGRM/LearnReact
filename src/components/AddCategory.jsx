import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

  /* uso de estados para cambiar el estado del input */

    const [inputValue, setInputValue] = useState('');

    /* funciiones para controlar con eventos los estados */

    const handleInputChange = (e) => {
      setInputValue(e.target.value);
    }

    /* funcion para el estadod e la categoria en el componente principal */

    const handleSubmit = (e) => {
      e.preventDefault();

      if(inputValue.length > 2){
        setCategories(cats => [ inputValue, ...cats ]);
        setInputValue('');
      }
      
      
    }

  return (
    <>
      <form onSubmit={handleSubmit}>
        <input 
            type="text" 
            value={inputValue}
            onChange={handleInputChange}
        />
      </form>
    </>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
