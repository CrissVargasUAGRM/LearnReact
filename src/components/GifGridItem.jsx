import React, { Fragment } from 'react'

export const GifGridItem = ({id, title, url}) => {
  return (
    <div className="card animate__animated animate__backInRight animate__delay-2s">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}
