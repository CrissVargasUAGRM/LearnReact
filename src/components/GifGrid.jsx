import React, { Fragment } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

     /* const [images, setImages] = useState([]); */

    /*1 const [count, setCount] = useState(0); */
    {/*1 <h3>{count}</h3>
        <button onClick={() => setCount(count + 1)}>Tocame ðŸ¥µ</button> */}

    const {data:images, loading} = useFetchGifs(category);

  return (
      <Fragment>
        <h3>{category}</h3>

        {loading ? 'Cargando...' : 'Data Cargada'}

        <div className='card-grid'>
            

            {images.map( (img) => {
                return (<GifGridItem 
                    key={img.id}
                    {...img}
                />)
            })}
        </div>
      </Fragment>
        
  )
}
