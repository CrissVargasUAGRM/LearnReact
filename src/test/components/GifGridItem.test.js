import React from 'react';
import {shallow} from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () => {

    const title = 'El titulo';
    const url = 'https://localhost/algo.jpg';

    const wrapper = shallow( <GifGridItem title={title} url={url}/> );

    const clase = 'animate__backInRight';

    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('debe de tener la imagen igual al url y alt de los props', () => { 
        const img = wrapper.find('img'); 
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
        
    });

    test('debe tener el espacio animate__backInRight', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes(clase)).toBe(true);
    });
    
    
});
