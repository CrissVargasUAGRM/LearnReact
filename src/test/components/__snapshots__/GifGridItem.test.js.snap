// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__backInRight animate__delay-2s"
>
  <img
    alt="El titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    El titulo
  </p>
</div>
`;
